Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.cpp

```diff
@@ -1083,13 +1164,150 @@ int Wiimote::CurrentExtension() const
   return m_extension->active_extension;
 }
 
-bool Wiimote::HaveExtension() const
+bool Wiimote::ExtensionLogic::ReadDeviceDetectPin()
 {
-  return m_extension->active_extension > 0;
+  return extension->active_extension ? true : false;
 }
 
-bool Wiimote::WantExtension() const
+void Wiimote::ExtensionLogic::Update()
 {
-  return m_extension->switch_extension != 0;
+  // Update controller data from user input:
+  // Write data to addr 0x00 of extension register
+  extension->GetState(reg_data.controller_data);
+}
+
+// TODO: move this to Common if it doesn't exist already?
+template <typename T>
+void SetBit(T& value, u32 bit_number, bool bit_value)
+{
+  bit_value ? value |= (1 << bit_number) : value &= ~(1 << bit_number);
+}
+
+void Wiimote::MotionPlusLogic::Update()
+{
+  if (!IsActive())
+  {
+    return;
+  }
+
+  // TODO: clean up this hackery:
+  // the value seems to increase based on time starting after the first read of 0x00
+  if (IsActive() && times_updated_since_activation < 0xff)
+  {
+    ++times_updated_since_activation;
+
+    // TODO: wtf is this value actually..
```

Comment:
The Switch joycons have a timer value that increments faster than 60hz and is included in the packets - probably the same kind of thing here?

Probably just use (getEmulatedTimeMillis() / 10) & 0xF or something.

Reply:
It's not just a timer. At least it's not behaving like one on real hardware. And Sports Resort disconnects the motion+ after reading it. I'm still figuring it out.